src/display/api.js

```diff
@@ -1818,8 +1817,18 @@ var WorkerTransport = (function WorkerTransportClosure() {
         switch (type) {
           case 'JpegStream':
             imageData = data[3];
-            loadJpegStream(id, imageData, pageProxy.objs);
-            break;
+            return new Promise((resolve, reject) => {
+              const img = new Image();
+              img.onload = function() {
+                resolve(img);
+              };
+              img.onerror = function() {
+                reject(new Error('Error during JPEG image loading'));
+              };
+              img.src = imageData;
+            }).then((img) => {
+              pageProxy.objs.resolve(id, img);
+            });
```

Comment:
In case of an error, `pageProxy.objs.resolve` is not called any more. Is that change intentional?

Reply:
Yes, this was intentional considering that when the native loading/decoding fails here we're now falling back to the built-in JPEG decoder. Hence we'll instead add the image to `pageProxy.objs` in the `case 'Image':` branch below, once the built-in decoding has happened.

Reply:
I see. Can you add a comment after `reject(new Error('Error during JPEG image loading'));` that briefly explains that the error is handled by the fallback logic in `core/evaluator.js`?

Reply:
Sure, a comment has been added; thank you!